mmocr/models/textrecog/backbones/resnet.py

```diff
@@ -0,0 +1,232 @@
+        self.use_plugins = False
+        self.arch_channels = arch_channels
+        self.res_layers = []
+        if plugins is not None:
+            self.plugin_ahead_names = []
+            self.plugin_after_names = []
+            self.use_plugins = True
+        for i, num_blocks in enumerate(arch_layers):
+            stride = strides[i]
+            channel = arch_channels[i]
+
+            if self.use_plugins:
+                self._make_stage_plugins(plugins, stage_idx=i)
+
+            res_layer = self._make_layer(
+                block_cfgs=block_cfgs,
+                inplanes=self.inplanes,
+                planes=channel,
+                blocks=num_blocks,
+                stride=stride,
+            )
+            self.inplanes = channel
+            layer_name = f'layer{i + 1}'
+            self.add_module(layer_name, res_layer)
+            self.res_layers.append(layer_name)
+
+    def _make_layer(self, block_cfgs, inplanes, planes, blocks, stride):
+        layers = []
+        downsample = None
+        block_cfgs_ = block_cfgs.copy()
+        if isinstance(stride, int):
+            stride = (stride, stride)
+
+        if stride[0] != 1 or stride[1] != 1 or inplanes != planes:
+            downsample = ConvModule(
+                inplanes,
+                planes,
+                1,
+                stride,
+                norm_cfg=dict(type='BN'),
+                act_cfg=None)
+
+        if block_cfgs_['type'] == 'BasicBlock':
+            block = BasicBlock
+            block_cfgs_.pop('type')
+        else:
+            raise ValueError('{} not implement yet'.format(block['type']))
+
+        layers.append(
+            block(
+                inplanes,
+                planes,
+                stride=stride,
+                downsample=downsample,
+                **block_cfgs_))
+        inplanes = planes
+        for _ in range(1, blocks):
+            layers.append(block(inplanes, planes, **block_cfgs_))
+
+        return Sequential(*layers)
+
+    def _make_stem_layer(self, in_channels, stem_channels):
+        if isinstance(stem_channels, int):
+            stem_channels = [stem_channels]
+        stem_layers = []
+        for _, channels in enumerate(stem_channels):
+            stem_layer = ConvModule(
+                in_channels,
+                channels,
+                kernel_size=3,
+                stride=1,
+                padding=1,
+                bias=False,
+                norm_cfg=dict(type='BN'),
+                act_cfg=dict(type='ReLU'))
+            in_channels = channels
+            stem_layers.append(stem_layer)
+        self.stem_layers = Sequential(*stem_layers)
+        self.inplanes = stem_channels[-1]
+
+    def _make_stage_plugins(self, plugins, stage_idx):
+        """Make plugins for ResNet ``stage_idx`` th stage.
+
+        Currently we support to insert ``nn.Maxpooling``,
+        ``nn.conv2d``into the backbone. This is designed for ResNet31
+        architecture likes
```

Comment:
```suggestion
        Currently we support inserting ``nn.Maxpooling``,
        ``nn.conv2d``into the backbone. Originally designed for
        ResNet31-like architectures.
```

Reply:
There is a strong need for me to retake Colleage-English course